docs/topics/metadata-jvm.md

```diff
@@ -0,0 +1,420 @@
+
+Here's an example where private functions are removed from a class's metadata:
+
+```kotlin
+// Imports the necessary libraries
+import kotlin.metadata.jvm.*
+import kotlin.metadata.*
+
+fun main() {
+    // Specifies the fully qualified class name
+    val className = "org.example.SampleClass"
+
+    try {
+        // Retrieves the class object for the specified name
+        val clazz = Class.forName(className)
+
+        // Retrieves the @Metadata annotation
+        val metadataAnnotation = clazz.getAnnotation(Metadata::class.java)
+        if (metadataAnnotation != null) {
+            println("Kotlin Metadata found for class: $className")
+
+            // Parses metadata using the readStrict() function
+            val metadata = KotlinClassMetadata.readStrict(metadataAnnotation)
+            if (metadata is KotlinClassMetadata.Class) {
+                val kmClass = metadata.kmClass
+
+                // Removes private functions from the class metadata
+                kmClass.functions.removeIf { it.visibility == Visibility.PRIVATE }
+                println("Removed private functions. Remaining functions: ${kmClass.functions.map { it.name }}")
+
+                // Serializes the modified metadata back
+                val newMetadata = metadata.write()
+                // After modifying the metadata, you need to write it into the class file
+                // To do so, you can use a bytecode manipulation framework such as ASM
+                
+                println("Modified metadata: ${newMetadata}")
+            } else {
+                println("The metadata is not a class.")
+            }
+        } else {
+            println("No Kotlin Metadata found for class: $className")
+        }
+    } catch (e: ClassNotFoundException) {
+        println("Class not found: $className")
+    } catch (e: Exception) {
+        println("Error processing metadata: ${e.message}")
+        e.printStackTrace()
+    }
+}
+```
+
+> Instead of separately calling `readStrict()` and `write()`, you can use the [`transform()`](https://kotlinlang.org/api/kotlinx-metadata-jvm/kotlin-metadata-jvm/kotlin.metadata.jvm/-kotlin-class-metadata/-companion/transform.html) function.
+> This function parses metadata, applies transformations through a lambda, and writes the modified metadata automatically.
+> 
+{style="tip"}
+
+## Create metadata from scratch
+
+To create metadata for a Kotlin class file from scratch using the Kotlin Metadata JVM library:
+
+1. Create an instance of `KmClass`, `KmPackage`, or `KmLambda`, depending on the type of metadata you want to generate.
+2. Populate the instance with attributes such as the class name, visibility, constructors, and function signatures.
+
+    > You can use Kotlin's `apply()` function to reduce boilerplate code while setting properties.
```

Comment:
Do we really need to specify that this is **Kotlin's** function?

Reply:
maybe it is better to say that this one is a scope function and provide a link to the Scope functions doc.